src/dtscalibration/datastore.py

```diff
@@ -1074,9 +1100,105 @@ def variance_stokes_exponential(
 
             return var_I, resid_da
 
+    def variance_stokes_linear(
+            self,
+            st_label,
+            sections=None,
+            nbin=50,
+            through_zero=True,
+            plot_fit=False):
+        """
+        Estimate a Stokes variance that is linear dependent on the intensity.
+
+        Parameters
+        ----------
+        st_label : str
+            Key under which the Stokes DataArray is stored. E.g., 'ST', 'REV-ST'
+        sections : dict, optional
+            Define sections. See documentation
+        nbin : int
+            Number of bins to compute the variance for, through which the linear
+            function is fitted. Make sure that that are at least 50 residuals
+            per
+            bin to compute the variance from.
+        through_zero : bool
+            If True, the variance is computed as: VAR(Stokes) = angle * Stokes
+            If False, VAR(Stokes) = angle * Stokes + offset.
+            From what we can tell from our inital trails, is that the offset
+            seems very small, so that True seems a better option.
+        plot_fit : bool
+            If True plot the variances for each bin and plot the fitted
+            linear function
+        """
+        import matplotlib.pyplot as plt
+
+        if sections:
+            self.sections = sections
+        else:
+            assert self.sections, 'sections are not defined'
+
+        st_var, resid = self.variance_stokes(st_label=st_label)
+
+        ix_sec = self.ufunc_per_section(x_indices=True, calc_per='all')
+        st = self.isel(x=ix_sec)[st_label].values.ravel()
+        diff_st = resid.isel(x=ix_sec).values.ravel()
+
+        # Adjust nbin silently to fit residuals in
+        # rectangular matrix and use numpy for computation
+        nbin_ = nbin
+        while st.size % nbin_:
+            nbin_ -= 1
+
+        if nbin_ != nbin:
+            print('Estimation of linear variance of', st_label,
+                  'Adjusting nbin to:', nbin_)
+            nbin = nbin_
+
+        isort = np.argsort(st)
+        st_sort_mean = st[isort].reshape((nbin, -1)).mean(axis=1)
+        st_sort_var = diff_st[isort].reshape((nbin, -1)).var(axis=1)
+
+        if through_zero:
+            # VAR(Stokes) = angle * Stokes
+            offset = 0.
+            angle = np.linalg.lstsq(st_sort_mean[:, None], st_sort_var,
+                                    rcond=None)[0]
+        else:
+            # VAR(Stokes) = angle * Stokes + offset
+            angle, offset = np.linalg.lstsq(
+                np.hstack((st_sort_mean[:, None], np.ones((nbin, 1)))),
+                st_sort_var,
+                rcond=None)[0]
+
+        var_fun = lambda stokes: angle * stokes + offset  # noqa: E731
```

Comment:
Wil je het niet aanpassen naar een functie def? ;)

volgens mij doet dit precies hetzelfde (en daarom is er E731):
def var_fun(stokes):
     return angle * stokes + offset